src/main/java/com/dzionek/simpleshell/commands/affecting/impl/CdCommand.java

```diff
@@ -0,0 +1,28 @@
+package com.dzionek.simpleshell.commands.affecting.impl;
+
+import com.dzionek.simpleshell.commands.affecting.SetterCommand;
+import com.dzionek.simpleshell.shellmanaging.ShellData;
+
+import java.io.File;
+import java.util.logging.FileHandler;
+
+/**
+ * @author Bartlomiej Janik
+ * @since 9/16/2017
+ */
+public class CdCommand implements SetterCommand {
+
+    StringBuilder stringBuilder = new StringBuilder();
+
+    public void executeSetter(ShellData shellData, String value) {
+        String tempPath = shellData.getCurrentDirectory();
+        if (value.equals("..")) {
+            shellData.setCurrentDirectory(tempPath.substring(0, tempPath.lastIndexOf("\\")));
+        } else {
+            stringBuilder.append(tempPath + "\\" + value);
+            if (new File(stringBuilder.toString()).isDirectory())
```

Comment:
jakbyś chciał to dopieścić to tego całego ifa który jest w else, wywalić do metody i użyć go tu, unikniesz jawnego if w if